test/e2e/specs/data-json.js

```diff
@@ -0,0 +1,250 @@
+      'description': 'It is an "Example".',
+      'timeMarks'  : {
+        'time'     : '12:18:29.000000',
+        'date_time': '2015-07-29 12:18:29.000000'
+      }
+    }
+
+    expect(updateResult.myJson).to.deep.include(expectedJson)
+
+    const getResult = await TestTable.findById(savedObj.objectId)
+
+    expect(getResult.myJson).to.deep.include(expectedJson)
+  })
+
+  it('array_append', async () => {
+    const jsonUpdateArrayAppend = jsonUpdateBuilder.ARRAY_APPEND()
+      .addArgument('$.decimals', 432.0)
+      .addArgument('$.colours', 'yellow')
+
+    const newValues = { myJson: jsonUpdateArrayAppend, objectId: savedObj.objectId }
+
+    const updateResult = await TestTable.save(newValues)
+
+    const expectedJson = {
+      'letter': 'a',
+      'number': 10,
+      'decimals': [12.3, 43.28, 56.89, 432.0],
+      'colours': ['red', 'green', 'blue', 'yellow'],
+      'description': 'It is an "Example".',
+      'timeMarks': {
+        'time': '12:18:29.000000',
+        'date': '2015-07-29',
+        'date_time': '2015-07-29 12:18:29.000000'
+      }
+    }
+
+    expect(updateResult.myJson).to.deep.include(expectedJson)
+
+    const getResult = await TestTable.findById(savedObj.objectId)
+
+    expect(getResult.myJson).to.deep.include(expectedJson)
+  })
+
+  it('array_insert', async () => {
+    const jsonUpdateArrayInsert = jsonUpdateBuilder.ARRAY_INSERT()
+      .addArgument('$.decimals[2]', 60)
+      .addArgument('$.colours[1]', 'cyan')
+
+    const newValues = { myJson: jsonUpdateArrayInsert, objectId: savedObj.objectId }
+
+    const updateResult = await TestTable.save(newValues)
+
+    const expectedJson = {
+      'letter': 'a',
+      'number': 10,
+      'decimals': [12.3, 43.28, 60, 56.89],
+      'colours': ['red', 'cyan', 'green', 'blue'],
+      'description': 'It is an "Example".',
+      'timeMarks': {
+        'time': '12:18:29.000000',
+        'date': '2015-07-29',
+        'date_time': '2015-07-29 12:18:29.000000'
+      }
+    }
+
+    expect(updateResult.myJson).to.deep.include(expectedJson)
+
+    const getResult = await TestTable.findById(savedObj.objectId)
+
+    expect(getResult.myJson).to.deep.include(expectedJson)
+  })
+
+  it('check toJSON() returns the same object as create()', async () => {
+    const dataByToJSON = jsonUpdateBuilder.SET()
+      .addArgument('$.letter', 'b')
+      .addArgument('$.number', 36)
+      .addArgument('$.state', true)
+      .addArgument('$.colours[0]', null)
+      .addArgument('$.innerObject', { a: 'b' })
+      .addArgument('$.innerArray', [4, 3, 2])
+      .toJSON()
+
+    const dataByCreate = jsonUpdateBuilder.SET()
+      .addArgument('$.letter', 'b')
+      .addArgument('$.number', 36)
+      .addArgument('$.state', true)
+      .addArgument('$.colours[0]', null)
+      .addArgument('$.innerObject', { a: 'b' })
+      .addArgument('$.innerArray', [4, 3, 2])
+      .create()
+
+    expect(dataByToJSON).to.deep.eql(dataByCreate)
```

Comment:
use only one instance 

and then:
`expect(dataByToJSON.toJSON()).to.deep.eql(dataByCreate.create())`

Reply:
+